src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,34 +36,111 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+//        this.value = address;
+        assignAddressComponents(address);
+    }
+    
+    /**
+     * Assign the address components into its individual classes.
+     * Only takes in the following address format a/BLOCK, STREET, UNIT, POSTAL_CODE
+     *
+     * @param string containing the entire address
+     */
+    private void assignAddressComponents(String address) throws IllegalValueException {
+    	final String[] addressComponents = address.split(ADDRESS_DELIMITER);
+    	
+    	if (addressComponents.length < PERSON_DATA_ADDRESS_POSTAL_CODE + 1) {
+            throw new IllegalValueException(MESSAGE_INVALID_ADDRESS);
+        }
+    	
+        addressBlock = new AddressBlock(addressComponents[PERSON_DATA_ADDRESS_BLOCK]);
+        addressStreet = new AddressStreet(addressComponents[PERSON_DATA_ADDRESS_STREET]);
+        addressUnit = new AddressUnit(addressComponents[PERSON_DATA_ADDRESS_UNIT]);
+        addressPostalCode = new AddressPostalCode(addressComponents[PERSON_DATA_ADDRESS_POSTAL_CODE]);
     }
 
     /**
-     * Returns true if a given string is a valid person email.
+     * Returns true if a given string is a valid person address.
      */
     public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
 
     @Override
     public String toString() {
-        return value;
+//        return value;
```

Comment:
No need to keep unused code, just delete it. If we need the code in the future we can revert back to previous version using git.